pkg/http/listener.go

```diff
@@ -207,9 +206,8 @@ func (listener *httpListener) start() {
 		}
 
 		if listener.errorLogFunc != nil {
-			listener.errorLogFunc(errors.New("junk message"),
-				"Received non-HTTP message from new connection %s at server %s",
-				bufconn.RemoteAddr(), bufconn.LocalAddr())
+			ctx := logger.SetContext(context.Background(), (&logger.ReqInfo{}).AppendTags("remoteAddr", bufconn.RemoteAddr().String()).AppendTags("localAddr", bufconn.LocalAddr().String()))
+			listener.errorLogFunc(ctx, err)
```

Comment:
IMO this is not the right way to fix logging for packages.  As `errorLogFunc` is a callback, package users should wrap any external requirements in his own and supply `errorLogFunc` accordingly.  Here we do not take advantage of callback functions. @